pkg/apis/networking/types.go

```diff
@@ -48,31 +48,35 @@ const (
 
 // NetworkPolicySpec provides the specification of a NetworkPolicy
 type NetworkPolicySpec struct {
-	// Selects the pods to which this NetworkPolicy object applies. The array of
-	// ingress rules is applied to any pods selected by this field. Multiple network
-	// policies can select the same set of pods. In this case, the ingress rules for
-	// each are combined additively. This field is NOT optional and follows standard
+	// Selects the pods to which this NetworkPolicy object applies for ingress and/or egress
+	// (as determined by `policyTypes`).  The relevant one or ones of `ingress` and `egress`
+	// list connections that this policy allows to be made from and to the selected pods.
+	// Multiple NetworkPolicy objects can select a given pod, and the connections they
+	// allow are combined additively.
+	// If any NetworkPolicy applies to a given pod for ingress then (a) the default is
+	// to allow ingress to the pod only from the pod's node and (b) the applicable policies
+	// add allowed ingress to the pod; otherwise all ingress to the pod is allowed.
+	// If any NetworkPolicy applies to a given pod for egress then (a) the default is
+	// to deny all egress from the pod and (b) the applicable policies add allowed egress from the pod;
+	// otherwise all egress from the pod is allowed.
+	// A connection from some source pod to some destination pod is allowed if egress from the source
+	// is allowed and ingress to the destination is allowed.
+	// This field is NOT optional and follows standard
 	// label selector semantics. An empty podSelector matches all pods in this
 	// namespace.
 	PodSelector metav1.LabelSelector
 
-	// List of ingress rules to be applied to the selected pods. Traffic is allowed to
-	// a pod if there are no NetworkPolicies selecting the pod
-	// (and cluster policy otherwise allows the traffic), OR if the traffic source is
-	// the pod's local node, OR if the traffic matches at least one ingress rule
-	// across all of the NetworkPolicy objects whose podSelector matches the pod. If
-	// this field is empty then this NetworkPolicy does not allow any traffic (and serves
-	// solely to ensure that the pods it selects are isolated by default)
+	// List of ingress rules to be applied to the selected pods if `policyTypes` includes "Ingress".
```

Comment:
Rather than mention policyTypes throughout, it might be simpler to phrase it something like this:

> Ingress is used only for policies whose policyTypes includes `Ingress`. For policies that do not have Ingress specified in their policyTypes, this field has no effect.
> List of ingress rules to be applied to the selected pods. If this field is empty then this network policy does not add any allowed ingress and serves solely to ensure that the pods it selects are isolated for ingress by default.